pkg/metrics/metrics.go

```diff
@@ -58,9 +64,37 @@ func newHistogramVec(name, help string, labels ...string) *prometheus.HistogramV
 // RegisterCustomMetrics registers the custom metrics
 func RegisterCustomMetrics() {
 	Reg = prometheus.NewRegistry()
+	if len(allHistogramVecs) == 0 {
+		log.Info("No Histograms to register")
+	}
 	// register metrics
 	for _, v := range allHistogramVecs {
 		Reg.MustRegister(v)
 	}
 	log.Info("custom metrics registered")
 }
+
+//nolint:unparam
+func prometheusHandler(ctx echo.Context) error {
+	h := promhttp.HandlerFor(Reg, promhttp.HandlerOpts{DisableCompression: true, Registry: Reg})
+	h.ServeHTTP(ctx.Response().Writer, ctx.Request())
+	return nil
+}
+
+func StartMetricsServer() *http.Server {
+	// start server
+	router := echo.New()
```

Comment:
`router` looks like a copy-paste zombie from the proxy code :-)

Reply:
fixed